test/e2e/parallel-runner.js

```diff
@@ -189,12 +196,45 @@ if (args.n) {
   process.exit()
 }
 
-concurrently(testcafeRuns, {
-  maxProcesses,
-  killOthers,
-}).then(
-  () => {},
-  () => {
+const checkCircleSha = async () => {
+  const { CIRCLECI, CIRCLE_BRANCH, CIRCLE_SHA1 } = process.env
+
+  if (E2E_BASE_URL && CIRCLECI && CIRCLE_BRANCH === 'master') {
+    const sleep = (ms = 0) => new Promise(resolve => setTimeout(resolve, ms))
+
+    let checkPing = true
+    const pingUrl = `${E2E_BASE_URL}/healthcheck/ping`
+    process.stdout.write(
+      `Checking ${pingUrl} for correct commit sha ${CIRCLE_SHA1}`
+    )
+
+    while (checkPing) {
+      try {
+        const { data } = await axios.get(pingUrl)
+
+        if (data.commit_id === CIRCLE_SHA1) {
+          checkPing = false
+          process.stdout.write(`${E2E_BASE_URL} returned correct commit sha`)
+        }
+      } catch (error) {
+        process.stdout.write('.')
+        await sleep(2000)
+      }
+    }
+  }
+}
```

Comment:
I would probably put this in the circleCI workflow and outside the test.
After deployment wait until the new sha is confirmed to be deployed or fail the deploy.

Feels like the test could assert that the version is correct but should not have to wait until it is.